zerver/lib/actions.py

```diff
@@ -5155,13 +5157,27 @@ def do_get_streams(user_profile: UserProfile, include_public: bool=True,
             recipient_check = Q(id__in=[sub.recipient.type_id for sub in user_subs])
         if include_public:
             invite_only_check = Q(invite_only=False)
-
-        if include_subscribed and include_public:
+        if include_owner_subscribed and user_profile.is_bot:
+            assert user_profile.bot_owner is not None
+            owner_subs = get_stream_subscriptions_for_user(user_profile.bot_owner).filter(
+                active=True,
+            ).select_related('recipient')
+            owner_subscribed_check = Q(id__in=[sub.recipient.type_id for sub in owner_subs])
+
+        if (include_subscribed and include_owner_subscribed and include_public):
+            query = query.filter(recipient_check | owner_subscribed_check | invite_only_check)
+        elif include_subscribed and include_owner_subscribed:
+            query = query.filter(recipient_check | owner_subscribed_check)
+        elif include_public and include_owner_subscribed:
+            query = query.filter(invite_only_check | owner_subscribed_check)
+        elif include_subscribed and include_public:
             query = query.filter(recipient_check | invite_only_check)
         elif include_public:
             query = query.filter(invite_only_check)
         elif include_subscribed:
             query = query.filter(recipient_check)
+        elif include_owner_subscribed:
+            query = query.filter(owner_subscribed_check)
```

Comment:
This combinatorial explosion of 2^N cases is definitely a bad idea.  It was kinda bad with 4 cases, but 8 is ridiculous.  I rewrote this to have N cases with OR gates, and will merge that version.

Reply:
Makes sense! I was gonna run this by you, but I think asking a question "in code" with a PR is faster than bothering you through PMs anyway. Thank you so much for fixing this, I wouldn't have thought of that! :)